packages/react/src/ReactContext.js

```diff
@@ -67,7 +67,61 @@ export function createContext<T>(
     $$typeof: REACT_PROVIDER_TYPE,
     _context: context,
   };
-  context.Consumer = context;
+
+  if (__DEV__) {
+    // A separate object, but proxies back to the original context object for
+    // backwards compatibility. It has a different $$typeof, so we can properly
+    // warn for the incorrect usage of Context as a Consumer.
+    const consumer = {
+      $$typeof: REACT_CONTEXT_TYPE,
+      _context: context,
+      Provider: context.Provider,
+    };
+    // $FlowFixMe: Flow complains about not setting a value, which is intentional here
+    Object.defineProperties(consumer, {
+      _calculateChangedBits: {
+        get() {
+          return context._calculateChangedBits;
+        },
+        set(_calculateChangedBits) {
+          context._calculateChangedBits = _calculateChangedBits;
+        },
+      },
+      _currentValue: {
+        get() {
+          return context._currentValue;
+        },
+        set(_currentValue) {
+          context._currentValue = _currentValue;
+        },
+      },
+      _currentValue2: {
+        get() {
+          return context._currentValue2;
+        },
+        set(_currentValue2) {
+          context._currentValue2 = _currentValue2;
+        },
+      },
+      Consumer: {
+        get() {
+          return context.Consumer;
```

Comment:
What if you render `<Context.Consumer.Consumer>`? Seems like that should also warn because it also relies on objects being shared.

Which makes me think: should the warning move into these getters instead? Fire for first getter accessed, ignore the rest. This could also nicely let us warn once per context type.

Reply:
Moving the warning into the getters will stop the error from coming up for just using `<Context />`. I'll address the nested `Consumer.Consumer` issue.

Reply:
Hi I'm getting the `Context.Consumer.Consumer` warning/error but I can't find where I am actually calling it in my code.

I just have 3 contexts and none of them are being used wrongly. May it be a bug or in the code that throws the warning or am I doing something extremely stupid?

I just wanted to let you know because I'm quite sure something is off here.